pkg/servicequota/vpcusage.go

```diff
@@ -0,0 +1,71 @@
+package servicequota
+
+import (
+	"fmt"
+
+	"github.com/oslokommune/okctl/pkg/config"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/servicequotas"
+	"github.com/logrusorgru/aurora"
+	"github.com/oslokommune/okctl/pkg/api/okctl.io/v1alpha1"
+)
+
+type vpcusage struct {
+	CloudProvider v1alpha1.CloudProvider
+}
+
+func (v vpcusage) Count() (int, error) {
+	vpcs, err := v.CloudProvider.EC2().DescribeVpcs(nil)
+	if err != nil {
+		return -1, fmt.Errorf("failed to get vpc count: %w", err)
+	}
+
+	return getLengthOf(getStringMapOf(vpcs.String()), "Vpcs")
+}
+
+func (v vpcusage) Quota() (int, error) {
+	quotas, err := v.CloudProvider.ServiceQuotas().GetServiceQuota(&servicequotas.GetServiceQuotaInput{
+		QuotaCode:   aws.String("L-F678F1CE"),
+		ServiceCode: aws.String("vpc"),
+	})
+	if err != nil {
+		return -1, fmt.Errorf("failed to get vpc quota: %w", err)
+	}
+
+	return getQuotaNumericValue(quotas)
+}
+
+func (vpcusage) Required() int {
+	return config.DefaultRequiredVpcs
+}
+
+func (vpcusage) Description() string {
+	return "You need a VPC to put your cluster in"
+}
+
+func (v vpcusage) CheckAvailability() error {
+	vpcQuota, err := v.Quota()
+	if err != nil {
+		return err
+	}
+
+	vpcCount, err := v.Count()
+	if err != nil {
+		return err
+	}
+
+	vpcAvail := vpcQuota - vpcCount
+	fmt.Printf("You have %d VPCs, and the quota for %s is %d,\n%d are available and you need %d", vpcCount, v.CloudProvider.Region(), vpcQuota, vpcAvail, v.Required())
```

Comment:
We should not be writing directly to `os.Stdout`, which is what `fmt.Printf` does. If we want to write, we should also receive an `io.Writer`, and then use `fmt.Fprintf` and family. This is preferable, because we can then pass in `ioutil.Discard`, which simply discards what we are writing, or even `os.Stderr`.

If you are passing in an `io.Writer` to the `NewVpcCheck`, then prefer to use the one configured in okctl, e.g., `o.Err`, also prefer err, since we don't want to clutter `os.Stdout` with log messages.

Reply:
Done :)